src/common/freeze/deepfreeze.ts

```diff
@@ -0,0 +1,21 @@
+
```

Comment:
I think is deep:

.frozen

Asserts that the target is frozen (cannot have new properties added to it and its existing properties cannot be modified).

var frozenObject = Object.freeze({});

expect(frozenObject).to.be.frozen;
expect({}).to.not.be.frozen;

From [chai](http://chaijs.com/api/bdd/)